reactor-test/src/main/java/reactor/test/StepVerifier.java

```diff
@@ -278,20 +282,27 @@ static void resetDefaultTimeout() {
 	 * to and verify. In order for operators to use virtual time, they must be invoked
 	 * from within the lambda.
 	 * @param options the verification options, including the supplier of the
-	 * {@link VirtualTimeScheduler} to inject and manipulate during verification.
+	 * {@link VirtualTimeScheduler} to inject and manipulate during verification
+	 * (see note above in case options doesn't define such a supplier)
 	 * @param <T> the type of the subscriber
 	 *
 	 * @return a builder for expectation declaration and ultimately verification.
 	 */
 	static <T> FirstStep<T> withVirtualTime(
 			Supplier<? extends Publisher<? extends T>> scenarioSupplier,
 			StepVerifierOptions options) {
+		//force the default VTS supplier if the provided options doesn't define a VTS supplier.
+		//note we make a copy just in case the original options are reused.
+		if (options.getVirtualTimeSchedulerSupplier() == null) {
+			options = options
+					.copy()
```

Comment:
What do you think about performing the copy operation after the verification of the input parameters?

Reply:
yeah that is indeed a good idea